### PR TITLE
:art: command refactoring

### PR DESCRIPTION
**Pull request by nothingelsematters**

### CODE REVIEW

**Comment by nothingelsematters** [comment]
Probable minus: a lot of unused variables in IDEA 🤔

File: src/main/kotlin/com/github/pool_party/pull_party_bot/commands/Command.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        val chatType = msg.chat.type
+        if ((chatType == "group" || chatType == "supergroup") &&
+            getChatAdministrators(chatId).join().all { it.user != sender }
+        ) {
+            sendMessage(chatId, ON_PERMISSION_DENY, "Markdown")
+            return@newCommand
+        }
+
+        action(msg, args)
+    }
```
Comment (ShuffleZZZ) [inline src/main/kotlin/com/github/pool_party/pull_party_bot/commands/Command.kt:61]: I wonder if it is possible to move all `new...` functions to class, maybe even as a constructors

File: src/main/kotlin/com/github/pool_party/pull_party_bot/commands/Command.kt [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    fun toBotCommand() = BotCommand(command, description)
+
+    companion object {
+        val all = mutableListOf<Command>()
+        val helpMessages = mutableMapOf<String, String>()
+    }
+}
+
+fun newCommand(commandName: String, description: String, helpMessage: String, action: Bot.(Message, String?) -> Unit) =
+    Command("/" + commandName, description, helpMessage, action)
```
Comment (ShuffleZZZ) [inline src/main/kotlin/com/github/pool_party/pull_party_bot/commands/Command.kt:33]: "/$commandName" IDEA suggestion

File: src/main/kotlin/com/github/pool_party/pull_party_bot/commands/PingCommandHandlers.kt [hunk 3]
```diff
@@ -174,13 +152,13 @@ private fun Bot.handleAdminsParty(msg: Message): String? {
 /**
  * Delete given parties from DataBase.
  */
-fun Bot.handleDelete(msg: Message, args: String?) {
+val delete = newCommand("delete", "forget the parties as they have never happened", HELP_DELETE) { msg, args ->
```
Comment (ShuffleZZZ) [inline src/main/kotlin/com/github/pool_party/pull_party_bot/commands/PingCommandHandlers.kt:155]: Supposed to be admin command

File: src/main/kotlin/com/github/pool_party/pull_party_bot/commands/PingCommandHandlers.kt [hunk 4]
```diff
@@ -198,7 +176,7 @@ fun Bot.handleDelete(msg: Message, args: String?) {
 /**
  * Delete all the parties of the chat.
  */
-fun Bot.handleClear(msg: Message) {
+val clear = newNoArgumentCommand("clear", "shut down all the parties ever existed", HELP_CLEAR) { msg ->
```
Comment (ShuffleZZZ) [inline src/main/kotlin/com/github/pool_party/pull_party_bot/commands/PingCommandHandlers.kt:179]: Same as `delete`

File: src/main/kotlin/com/github/pool_party/pull_party_bot/commands/Command.kt (hunk 1 above)
Comment (nothingelsematters) [reply src/main/kotlin/com/github/pool_party/pull_party_bot/commands/Command.kt:61]: Believe me, I've tried. The code is sick as fuck then

Now:
```kotlin
val command = newCommand("name", "desc", "help") { 
    // some shit to do
}
```

And with constructor:
```kotlin
val command = Command("name", "desc", "help", command@ {
    // some shit to do
    if (condition) return@command
    // do something more
})
```

Kotlin thinks that constructor is not a function, apparently 🥴🥴🥴

File: src/main/kotlin/com/github/pool_party/pull_party_bot/commands/PingCommandHandlers.kt (hunk 3 above)
Comment (nothingelsematters) [reply src/main/kotlin/com/github/pool_party/pull_party_bot/commands/PingCommandHandlers.kt:155]: Creator cannot delete it's own party, is that ok?

**Comment by nothingelsematters** [comment]
@ShuffleZZZ @Shuffle45 kkPowerful

File: src/main/kotlin/com/github/pool_party/pull_party_bot/commands/PingCommandHandlers.kt (hunk 3 above)
Comment (ShuffleZZZ) [reply src/main/kotlin/com/github/pool_party/pull_party_bot/commands/PingCommandHandlers.kt]: We don't store the information about creator, so how would yuo do that?

**Review by ShuffleZZZ — APPROVED** [review]

**Pull request merged**